File: Design-Patterns/StateMachine/Cpp/StateMachine/menu_scene.cpp
#include "scene.h"

menu_scene::menu_scene(scene_controller* controller) : scene(controller)
{
	
}

void menu_scene::start()
{
	std::cout << "Menu scene started" << std::endl;
}

void menu_scene::update()
{
	std::cout << "Menu scene updated" << std::endl;
}

void menu_scene::end()
{
	std::cout << "Menu scene ended" << std::endl;
}

File: Design-Patterns/StateMachine/CSharp/StateMachine/Scenes/MenuScene.cs
using System;

namespace StateMachine.Scenes
{
    class MenuScene : Scene
    {
        public MenuScene(SceneController controller) : base(controller)
        {

        }

        public override void Start()
        {
            Console.WriteLine("Menu scene started");
        }

        public override void Update()
        {
            Console.WriteLine("Menu scene updated");
        }

        public override void End()
        {
            Console.WriteLine("Menu scene ended");
        }
    }
}
